File: BloodBankFrontend/bloodbank/src/services/PatientService.js
import authAxios from "./authAxios";

const API_URL = "/api/patients";

class PatientService {
  getAllPatients() {
    return authAxios.get(API_URL);
  }

  getPatientById(id) {
    return authAxios.get(`${API_URL}/${id}`);
  }

  createPatient(patient) {
    return authAxios.post(API_URL, patient);
  }

  updatePatient(id, patient) {
    return authAxios.put(`${API_URL}/${id}`, patient);
  }

  deletePatient(id) {
    return authAxios.delete(`${API_URL}/${id}`);
  }
}

const patientService = new PatientService();
export default patientService;

File: BloodBankBackend/src/main/java/com/itvedant/bloodbankmanagement/service/PatientService.java
package com.itvedant.bloodbankmanagement.service;

import com.itvedant.bloodbankmanagement.entity.Patient;
import java.util.List;

public interface PatientService {
    List<Patient> getAllPatients();
    Patient getPatientById(Long id);
    Patient createPatient(Patient patient);
    Patient updatePatient(Long id, Patient patientDetails);
    void deletePatient(Long id);
}

